src/ring_theory/matrix_equiv_tensor.lean

```diff
@@ -0,0 +1,197 @@
+λ i j, a * algebra_map R A (m i j)
+
+/--
+(Implementation detail).
+The function underlying `(A ⊗[R] matrix n n R) →ₐ[R] matrix n n A`,
+as an `R`-linear map in the second tensor factor.
+-/
+def to_fun_right_linear (a : A) : matrix n n R →ₗ[R] matrix n n A :=
+{ to_fun := to_fun R A n a,
+  map_add' := λ x y, by { dsimp only [to_fun], ext, simp [mul_add], },
+  map_smul' := λ r x,
+  begin
+    dsimp only [to_fun],
+    ext,
+    simp only [matrix.smul_val, pi.smul_apply, ring_hom.map_mul,
+      algebra.id.smul_eq_mul, ring_hom.map_mul],
+    rw [algebra.smul_def r, mul_left_comm],
+  end, }
+
+/--
+(Implementation detail).
+The function underlying `(A ⊗[R] matrix n n R) →ₐ[R] matrix n n A`,
+as an `R`-bilinear map.
+-/
+def to_fun_bilinear : A →ₗ[R] matrix n n R →ₗ[R] matrix n n A :=
+{ to_fun := to_fun_right_linear R A n,
+  map_add' := λ x y, by { ext, simp [to_fun_right_linear, to_fun, add_mul], },
+  map_smul' := λ r x, by { ext, simp [to_fun_right_linear, to_fun] }, }
+
+/--
+(Implementation detail).
+The function underlying `(A ⊗[R] matrix n n R) →ₐ[R] matrix n n A`,
+as an `R`-linear map.
+-/
+def to_fun_linear : A ⊗[R] matrix n n R →ₗ[R] matrix n n A :=
+tensor_product.lift (to_fun_bilinear R A n)
+
+/--
+The function `(A ⊗[R] matrix n n R) →ₐ[R] matrix n n A`, as an algebra homomorphism.
+-/
+def to_fun_alg_hom : (A ⊗[R] matrix n n R) →ₐ[R] matrix n n A :=
+alg_hom_of_linear_map_tensor_product (to_fun_linear R A n)
+begin
+  intros, ext,
+  simp only [to_fun_linear, to_fun_bilinear, to_fun_right_linear, to_fun, lift.tmul,
+    linear_map.coe_mk, matrix.mul_mul_left, matrix.smul_val, matrix.mul_eq_mul,
+    matrix.mul_mul_right, ring_hom.map_matrix_mul],
+  rw [←_root_.mul_assoc, mul_comm a₁ a₂],
+end
+begin
+  intros, ext,
+  simp only [to_fun_linear, to_fun_bilinear, to_fun_right_linear, to_fun, matrix.one_val,
+    algebra_matrix_matrix_val, lift.tmul, linear_map.coe_mk],
+  split_ifs; simp,
+end
+
+@[simp] lemma to_fun_alg_hom_apply (a : A) (m : matrix n n R) :
+  to_fun_alg_hom R A n (a ⊗ₜ m) = λ i j, a * algebra_map R A (m i j) :=
+begin
+  simp [to_fun_alg_hom, alg_hom_of_linear_map_tensor_product, to_fun_linear],
+  refl,
+end
+
+/--
+(Implementation detail.)
+
+The bare function `matrix n n A → A ⊗[R] matrix n n R`.
+(We don't need to show that it's an algebra map, thankfully --- just that it's an inverse.)
+-/
+def inv_fun (M : matrix n n A) : A ⊗[R] matrix n n R :=
+∑ (p : n × n), M p.1 p.2 ⊗ₜ (λ i' j', if (i', j') = p then 1 else 0)
+
+@[simp] lemma inv_fun_zero : inv_fun R A n 0 = 0 :=
+by simp [inv_fun]
+
+@[simp] lemma inv_fun_add (M N : matrix n n A) :
+  inv_fun R A n (M + N) = inv_fun R A n M + inv_fun R A n N :=
+by simp [inv_fun, add_tmul, finset.sum_add_distrib]
+
+
+@[simp] lemma inv_fun_smul (a : A) (M : matrix n n A) :
+  inv_fun R A n (λ i j, a * M i j) = (a ⊗ₜ 1) * inv_fun R A n M :=
+by simp [inv_fun,finset.mul_sum]
+
+-- I don't think it is so generally useful it belongs in `algebra.lean`.
```

Comment:
Why not? That seems like the right place to me...

Reply:
Sure.